src/main/java/pl/com/itsystems/springmvc/HomeController.java

```diff
@@ -0,0 +1,63 @@
+package pl.com.itsystems.springmvc;
+
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+import java.util.UUID;
+
+@Controller
+public class HomeController {
+    private ProductRepository productRepository;
+
+    public HomeController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping("/")
+    public String homePage(Model model){
+        List<Product> productList = productRepository.findAll();
+        model.addAttribute("products", productList);
+        return "index";
+    }
+
+    @GetMapping("/list")
+    public String getCategory(String kategoria, Model model){
+        double totalPrice = 0;
+        List<Product> productList = productRepository.findByCategory(ProductCategory.valueOf(kategoria.toUpperCase()));
+        for (Product product : productList) {
+            totalPrice += product.getPrice();
+        }
```

Comment:
Rozważyłbym wprowadzenie dodatkowej klasy ProductService, aby nie robić obliczeń w kontrolerze. To nieobowiązkowe